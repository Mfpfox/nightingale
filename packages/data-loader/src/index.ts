import { defineElement } from "@nightingale-elements/utils";
import DataLoader from "./data-loader";

defineElement(DataLoader);
