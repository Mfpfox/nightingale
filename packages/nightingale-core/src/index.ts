import NightingaleBaseElement from "./nightingale-base-element";
import _withZoom from "./withZoom";
import _withDimensions from "./withDimensions";
import _withPosition from "./withPosition";
import _withMargin from "./withMargin";

export const withZoom = _withZoom;
export const withDimensions = _withDimensions;
export const withPosition = _withPosition;
export const withMargin = _withMargin;

export default NightingaleBaseElement;
