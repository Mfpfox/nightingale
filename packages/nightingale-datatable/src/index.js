import { defineElement } from "@nightingale-elements/utils";
import NightingaleDatatable from "./nightingale-datatable";

defineElement(NightingaleDatatable);

export default NightingaleDatatable;
