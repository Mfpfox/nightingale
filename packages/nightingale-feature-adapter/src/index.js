import { defineElement } from "@nightingale-elements/utils";
import NightingaleFeatureAdapter from "./nightingale-feature-adapter";

defineElement(NightingaleFeatureAdapter);

export default NightingaleFeatureAdapter;
export {
  formatTooltip,
  renameProperties,
  formatXrefs,
  getEvidenceFromCodes,
} from "./BasicHelper";
