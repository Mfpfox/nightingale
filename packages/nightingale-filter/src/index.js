import { defineElement } from "@nightingale-elements/utils";
import NightingaleFilter from "./nightingale-filter";

defineElement(NightingaleFilter);

export default NightingaleFilter;
