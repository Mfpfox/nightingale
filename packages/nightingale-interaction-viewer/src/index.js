import { defineElement } from "@nightingale-elements/utils";
import NightingaleInteractionViewer from "./nightingale-interaction-viewer";

defineElement(NightingaleInteractionViewer);

export default NightingaleInteractionViewer;
