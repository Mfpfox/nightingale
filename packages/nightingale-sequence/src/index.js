import { defineElement } from "@nightingale-elements/utils";
import NightingaleSequence from "./nightingale-sequence";

defineElement(NightingaleSequence);

export default NightingaleSequence;
