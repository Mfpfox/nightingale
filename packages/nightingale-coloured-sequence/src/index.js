import { defineElement } from "@nightingale-elements/nightingale-core";
import NightingaleColouredSequence from "./nightingale-coloured-sequence";

defineElement(NightingaleColouredSequence);

export default NightingaleColouredSequence;
