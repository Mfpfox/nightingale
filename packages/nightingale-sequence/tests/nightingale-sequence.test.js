import "../src/index";

let rendered;

describe("nightingale-sequence tests", () => {
  beforeEach(() => {
    document.documentElement.innerHTML = `<nightingale-sequence sequence="MAMYDDEFDTKASDL" length="15"></nightingale-sequence>`;
    rendered = document.querySelector("nightingale-sequence");
  });

  afterEach(() => {
    document.querySelector("nightingale-sequence").remove();
  });

  test("it should display the sequence correctly", async () => {
    expect(rendered).toMatchSnapshot();
  });

  test("it should display the sequence correctly after zoom", (done) => {
    rendered.setAttribute("displaystart", "2");
    rendered.setAttribute("displayend", "4");
    window.requestAnimationFrame(() => {
      expect(rendered).toMatchSnapshot();
      done();
    });
  });

  test("it should display the sequence correctly after highlight", (done) => {
    rendered.setAttribute("highlight", "2:5");
    window.requestAnimationFrame(() => {
      expect(rendered).toMatchSnapshot();
      done();
    });
  });
});
