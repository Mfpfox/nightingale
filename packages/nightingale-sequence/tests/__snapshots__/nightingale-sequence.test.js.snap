// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nightingale-sequence tests it should display the sequence correctly 1`] = `
<nightingale-sequence
  length="15"
  sequence="MAMYDDEFDTKASDL"
  style="display: block; width: 100%;"
>
  <div
    class=""
  >
    <svg
      height="44"
      id=""
      width="100"
    >
      <g
        class="background"
      />
      <g
        class="x axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(3.3333333333333335,0)"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(27.166666666666668,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(60.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(93.83333333333333,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            15
          </text>
        </g>
      </g>
      <g
        class="sequence"
        style="opacity: -1.333333333333333;"
        transform="translate(0,33)"
      />
      <g
        class="highlighted"
      >
        <rect
          fill="rgba(255, 235, 59, 0.8)"
          height="44"
          style="opacity: 0.5; pointer-events: none;"
          width="0"
          x="0"
        />
      </g>
    </svg>
  </div>
</nightingale-sequence>
`;

exports[`nightingale-sequence tests it should display the sequence correctly after highlight 1`] = `
<nightingale-sequence
  highlightstart="4"
  length="15"
  sequence="MAMYDDEFDTKASDL"
  style="display: block; width: 100%;"
>
  <div
    class=""
  >
    <svg
      height="44"
      id=""
      width="100"
    >
      <g
        class="background"
      />
      <g
        class="x axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(3.3333333333333335,0)"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(27.166666666666668,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(60.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(93.83333333333333,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            15
          </text>
        </g>
      </g>
      <g
        class="sequence"
        style="opacity: -1.333333333333333;"
        transform="translate(0,33)"
      />
      <g
        class="highlighted"
      >
        <rect
          fill="rgba(255, 235, 59, 0.8)"
          height="44"
          style="opacity: 0.5; pointer-events: none;"
          width="26.666666666666668"
          x="0"
        />
      </g>
    </svg>
  </div>
</nightingale-sequence>
`;

exports[`nightingale-sequence tests it should display the sequence correctly after zoom 1`] = `
<nightingale-sequence
  displayend="4"
  displaystart="2"
  length="15"
  sequence="MAMYDDEFDTKASDL"
  style="display: block; width: 100%;"
>
  <div
    class=""
  >
    <svg
      height="44"
      id=""
      width="100"
    >
      <g
        class="background"
      >
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="44"
          width="33.33333333333333"
          x="-33.33333333333333"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="44"
          width="33.33333333333333"
          x="0"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="44"
          width="33.33333333333333"
          x="33.33333333333333"
        />
        <rect
          class="base_bg feature"
          fill="#eee"
          height="44"
          width="33.33333333333333"
          x="66.66666666666666"
        />
        <rect
          class="base_bg feature"
          fill="#ccc"
          height="44"
          width="33.33333333333333"
          x="100"
        />
      </g>
      <g
        class="x axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(16.666666666666664,0)"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            2
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(33.83333333333333,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            3
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(67.16666666666666,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            4
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(100.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            5
          </text>
        </g>
      </g>
      <g
        class="sequence"
        style="opacity: 1;"
        transform="translate(0,33)"
      >
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="-16.666666666666664"
        >
          M
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="16.666666666666664"
        >
          A
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="49.99999999999999"
        >
          M
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="83.33333333333331"
        >
          Y
        </text>
        <text
          class="base"
          style="pointer-events: none; font-family: monospace;"
          text-anchor="middle"
          x="116.66666666666666"
        >
          D
        </text>
      </g>
      <g
        class="highlighted"
      >
        <rect
          fill="rgba(255, 235, 59, 0.8)"
          height="44"
          style="opacity: 0.5; pointer-events: none;"
          width="0"
          x="-33.33333333333333"
        />
      </g>
    </svg>
  </div>
</nightingale-sequence>
`;
